replace deprecated semver.Max()

lint complaining about deprecated semver.Max
```
❯ make clean && make && make test
...
= make lint =============================================================
golangci/golangci-lint info checking GitHub for tag 'v1.30.0'
golangci/golangci-lint info found version: 1.30.0 for v1.30.0/linux/amd64
golangci/golangci-lint info installed out/linters/golangci-lint
hack/update/github.go:231:14: SA1019: semver.Max is deprecated: use Compare instead. In most cases, returning a canonicalized version is not expected or desired.  (staticcheck)
                                stable = semver.Max(ver, stable)
                                         ^
hack/update/github.go:233:14: SA1019: semver.Max is deprecated: use Compare instead. In most cases, returning a canonicalized version is not expected or desired.  (staticcheck)
                                latest = semver.Max(ver, latest)
                                         ^
make[1]: *** [Makefile:469: lint-ci] Error 1
...
```

--- a/hack/update/github.go
+++ b/hack/update/github.go
@@ -228,9 +228,13 @@ func GHReleases(ctx context.Context, owner, repo string) (stable, latest string,
 			// check if ver version is release (ie, 'v1.19.2') or pre-release (ie, 'v1.19.3-rc.0' or 'v1.19.0-beta.2')
 			prerls := semver.Prerelease(ver)
 			if prerls == "" {
-				stable = semver.Max(ver, stable)
+				if semver.Compare(ver, stable) == 1 {
+					stable = ver
+				}
 			} else if strings.HasPrefix(prerls, "-rc") || strings.HasPrefix(prerls, "-beta") {
-				latest = semver.Max(ver, latest)
+				if semver.Compare(ver, latest) == 1 {
+					latest = ver
+				}
 			}
 			// make sure that latest >= stable
 			if semver.Compare(latest, stable) == -1 {

